src/tty/unix.rs

```diff
@@ -147,228 +441,47 @@ impl PosixRawReader {
-                if seq5.is_digit(10) {
-                    let seq6 = self.next_char()?;
-                    if seq6.is_digit(10) {
-                        self.next_char()?; // 'R' expected
-                    } else if seq6 == 'R' {
-                    } else {
-                        debug!(target: "rustyline",
-                               "unsupported esc sequence: ESC [ {}{} ; {} {}", seq2, seq3, seq5, seq6);
-                    }
-                } else {
-                    debug!(target: "rustyline",
-                           "unsupported esc sequence: ESC [ {}{} ; {:?}", seq2, seq3, seq5);
-                }
-                Ok(KeyPress::UnknownEscSeq)
-            } else if seq4.is_digit(10) {
-                let seq5 = self.next_char()?;
-                if seq5 == '~' {
-                    Ok(match (seq2, seq3, seq4) {
-                        ('2', '0', '0') => KeyPress::BracketedPasteStart,
-                        ('2', '0', '1') => KeyPress::BracketedPasteEnd,
-                        _ => {
-                            debug!(target: "rustyline",
-                                   "unsupported esc sequence: ESC [ {}{}{}~", seq2, seq3, seq4);
-                            KeyPress::UnknownEscSeq
-                        }
-                    })
-                } else {
-                    debug!(target: "rustyline",
-                           "unsupported esc sequence: ESC [ {}{}{} {}", seq2, seq3, seq4, seq5);
-                    Ok(KeyPress::UnknownEscSeq)
-                }
-            } else {
-                debug!(target: "rustyline",
-                       "unsupported esc sequence: ESC [ {}{} {:?}", seq2, seq3, seq4);
-                Ok(KeyPress::UnknownEscSeq)
-            }
-        } else if seq3 == ';' {
-            let seq4 = self.next_char()?;
-            if seq4.is_digit(10) {
-                let seq5 = self.next_char()?;
-                if seq5.is_digit(10) {
-                    self.next_char()?; // 'R' expected
-                    Ok(KeyPress::UnknownEscSeq)
-                } else if seq2 == '1' {
-                    Ok(match (seq4, seq5) {
-                        ('5', 'A') => KeyPress::ControlUp,
-                        ('5', 'B') => KeyPress::ControlDown,
-                        ('5', 'C') => KeyPress::ControlRight,
-                        ('5', 'D') => KeyPress::ControlLeft,
-                        ('2', 'A') => KeyPress::ShiftUp,
-                        ('2', 'B') => KeyPress::ShiftDown,
-                        ('2', 'C') => KeyPress::ShiftRight,
-                        ('2', 'D') => KeyPress::ShiftLeft,
-                        _ => {
-                            debug!(target: "rustyline",
-                                   "unsupported esc sequence: ESC [ 1 ; {} {:?}", seq4, seq5);
-                            KeyPress::UnknownEscSeq
+                EscapeSearchResult::IsPrefix => {}
+                EscapeSearchResult::NoMatch => {
```

Comment:
I am afraid we need to read the complete escape sequence before returning `UnknownEscSeq`.
Otherwise, we will print its tail.
I guess, we should better use something like `vte` crate.